rust/src/ftp/response.rs

```diff
@@ -0,0 +1,196 @@
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::ptr;
+use std::slice;
+
+#[repr(C)]
+pub struct FTPResponseLine {
+    code: *mut c_char,     // Response code as a string (may be null)
```

Comment:
Shouldn't we treat all these as u8? We generally treat even text input as byte arrays